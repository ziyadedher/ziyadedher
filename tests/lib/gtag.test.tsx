/**
 * @jest-environment jsdom
 */

// eslint-disable-next-line import/no-unassigned-import -- @testing-library/jest-dom doesn't need to be bound.
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import GTag, { pageview } from "../../src/lib/gtag";

describe("gtag", () => {
  describe("component", () => {
    test("renders", () => {
      const result = render(<GTag />);
      expect(result).toMatchSnapshot();
    });
  });

  describe("pageview", () => {
    test("calls gtag", () => {
      window.gtag = jest.fn() as jest.MockedFunction<typeof window.gtag>;
      pageview("/example");
      expect(window.gtag).toHaveBeenCalledTimes(1);
      expect(window.gtag).toHaveBeenCalledWith("config", expect.any(String), {
        // eslint-disable-next-line @typescript-eslint/naming-convention, camelcase -- gtag convention.
        page_path: "/example",
      });
    });
  });
});
