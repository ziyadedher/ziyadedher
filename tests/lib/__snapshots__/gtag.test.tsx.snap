// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gtag component renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <script
      data-nscript="afterInteractive"
      src="https://www.googletagmanager.com/gtag/js?id=google_analytics_measurement_id"
    />
    <script
      data-nscript="afterInteractive"
      id="gtag-init"
    >
      
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', 'google_analytics_measurement_id', {
            page_path: window.location.pathname,
          });
         
    </script>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
